bugRzilla_review.Rmd

```diff
@@ -168,45 +215,659 @@ Also there are columns which are empty so they are of no use of the analysis lik
+#converting the required fields in the correct datatype format
 bugs_df <- bugs_df %>%
     mutate_at(vars("creation_ts", "delta_ts", "lastdiffed", "deadline"), as.Date)
+# Taking the columns which are useful
+bugs_df <- bugs_df %>%
+    select("bug_id", "bug_severity", "bug_status", "creation_ts", "delta_ts", "op_sys", "priority", "resolution", "component_id", "version", "lastdiffed", "deadline")
+#for quick view of the datatypes and the structure of data
+skim(bugs_df)
+#showing the `datatable`
+datatable(head(bugs_df, 5), options = list(scrollX = TRUE))
 ```
 
+### About the Bugs Data used for Analysis
+
+<font size="4">
+I've taken the 12 columns under consideration to Analyse the Data. The brief description about the columns as follows:
+<ol>
+    <li>
+        **bug_id: **Unique numeric identifier for bug.
+    </li>
+    <li>
+        **bug_severity: **How severe the bug is, e.g. enhancement, critical, etc.
+    </li>
+    <li>
+        **bug_status: **Current status, e.g. NEW, RESOLVED, etc.
+    </li>
+    <li>
+        **creation_ts: **When bug was filed.
+    </li>
+    <li>
+        **delta_ts: **The timestamp of the last update on the bug. This includes updates to some related tables (e.g. "longdescs").
+    </li>
+    <li>
+        **op_sys: **Operating system bug was seen on, e.g. Windows Vista, Linux, etc.
+    </li>
+    <li>
+        **priority: **The priority of the bug (P1 = most urgent, P5 = least urgent).
+    </li>
+    <li>
+        **resolution: **The resolution, if the bug is in a closed state, e.g. FIXED, DUPLICATE, etc.
+    </li>
+    <li>
+        **component_id: **Numeric ids of the components.
+    </li>
+    <li>
+        **version: **Version of software in which bug is seen.
+    </li>
+    <li>
+        **lastdiffed: **The time at which information about this bug changing was last emailed to the cc list.
+    </li>
+    <li>
+        **deadline: **Date by which bug must be fixed.
+    </li>
+</ol>
+</font>
+
 ### Visualizations
 
-```{r}
-# Line plot for creation_ts, delta_ts, deadline
-creation <- bugs_df$creation_ts
-delta <- bugs_df$delta_ts
-deadline <- bugs_df$deadline
-last_diffed <- bugs_df$lastdiffed
-bug_id <- bugs_df$bug_id
-component_id <- bugs_df$component_id
-resolution <- bugs_df$resolution
+```{r Bug created graph}
+bug_created <- bugs_df %>%
+    ggplot(aes(x = creation_ts, y = bug_id)) +
+    geom_line(color = "darkorchid4") +
+    labs(title = "Bug Creation",
+           subtitle = "The data frame is sent to the plot using pipes",
+           y = "Bug ID",
+           x = "Date") +
+    theme_bw(base_size = 15)
+ggplotly(bug_created)
+```
+<font size="4">
+    The `Time-series graph` shows that which bug_id was filed in which `month` and `year` and from the `bar graph` we can conclude that in which `year` the most bugs are filed and when one will zoom the graphs, one can see on which date which bug was filed. The most of the Bugs are filled in the month of `January` and `July`.
```

Comment:
The blips are there because at that particular period of time the bugs were created.

Reply:
Okay, but it would interesting to identify the IDs for these bugs and look at the records. You should at least note the bug IDs in your report as being unusual cases.